src/Compilers/VisualBasic/Portable/Lowering/LocalRewriter/LocalRewriter_Conversion.vb

```diff
@@ -1,20 +1,32 @@
 ﻿' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
 Imports System.Collections.Immutable
-Imports System.Diagnostics
-Imports System.Runtime.InteropServices
 Imports Microsoft.CodeAnalysis.PooledObjects
-Imports Microsoft.CodeAnalysis.Text
 Imports Microsoft.CodeAnalysis.VisualBasic.Symbols
-Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
-Imports TypeKind = Microsoft.CodeAnalysis.TypeKind
 
 Namespace Microsoft.CodeAnalysis.VisualBasic
     Partial Friend NotInheritable Class LocalRewriter
         Public Overrides Function VisitConversion(node As BoundConversion) As BoundNode
 
             If Not _inExpressionLambda AndAlso Conversions.IsIdentityConversion(node.ConversionKind) Then
-                Return Visit(node.Operand)
+
+                Dim result = DirectCast(Visit(node.Operand), BoundExpression)
+
+                If node.ExplicitCastInCode AndAlso IsFloatingPointExpressionOfUnknownPrecision(result) Then
+                    ' To force a value of a floating point type to the exact precision of its type, an explicit cast can be used.
+                    ' It means that explicit casts to CDbl() or CSng() should be preserved on the node.
+                    ' If original conversion has become something else with unknown precision, add an explicit identity cast.
+                    result = node.Update(
+                        result,
+                        node.ConversionKind,
```

Comment:
Should this be `ConversionKind.Identity` explicitly?

Reply:
I only managed to hit this path with identity conversion, but I did keep some handling for a hypothetical case (sequence) in `IsFloatingPointExpressionOfUnknownPrecision`. So I changed to use ConversionKind.Identity as you suggest, and also set the next two parameters (as they are in C#).

---
In reply to: [143078002](https://github.com/dotnet/roslyn/pull/22552#discussion_r143078002) [](ancestors = 143078002)